File: backend/internal/models/establishment.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Establishment struct {
	ID        uuid.UUID `gorm:"type:uuid;primaryKey"`
	Number    string    `gorm:"column:number"`
	Name      string    `gorm:"column:name"`
	LegalName string    `gorm:"column:legal_name"`
	AddressID uuid.UUID `gorm:"column:address_id"`
	Address   Address   `gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"` // Cascade não funciona (F)

	CreatedAt time.Time `gorm:"column:created_at"`
	UpdatedAt time.Time `gorm:"column:updated_at"`
}

File: frontend/src/types/establishment.ts
export type EstablishmentResponse = {
  ID: string;
  Number: string;
  Name: string;
  LegalName: string;
  AddressID: string;
  Address: {
    ID: string;
    Street: string;
    Number: string;
    City: string;
    State: string;
    ZipCode: string;
    CreatedAt: Date;
    UpdatedAt: Date;
  },
  CreatedAt: Date;
  UpdatedAt: Date;
}

export type EstablishmentPreview = {
  id: string;
  name: string;
  legal_name: string;
  number: string;
  address: {
    street: string;
    number: string;
    city: string;
    state: string;
    zip_code: string;
  }
}
